### PR TITLE
Bug 1

### PR DESCRIPTION
**Pull request by antonico13**

### CODE REVIEW

File: src/Controller/QuestionTemplateController.php [hunk 1]
```diff
@@ -171,9 +171,15 @@ public function editQuestion (RouteMatch $routeMatch, Request $request) {
          * @var $question QuestionTemplate
          */
         $question = $this->questionService->getQuestion($id);
+
         if ($question == null) {
```
Comment (CalinBolea) [inline src/Controller/QuestionTemplateController.php]: yoda style, strict comparison

File: src/Controller/QuizTemplateController.php [hunk 2]
```diff
@@ -175,10 +176,19 @@ public function editQuizzes (RouteMatch $routeMatch, Request $request) {
         }
 
         $id = $routeMatch->getRequestAttributes()['id'];
+        /**
+         * @var QuizTemplate $quiz
```
Comment (calinpristavu) [inline src/Controller/QuizTemplateController.php]: return type of `quizService->getQuiz()` should be `?QuizTemplate` not `?EntityInterface` 

if you make that change, this `@var` is no longer needed

File: src/Controller/QuestionTemplateController.php [hunk 3]
```diff
@@ -171,9 +171,15 @@ public function editQuestion (RouteMatch $routeMatch, Request $request) {
...
             return $this->renderer->renderException(['message' => 'Question not found'], 404);
         }
+
+        if ($question->getUser()->getID() != $this->session->get('id')) {
```
Comment (Raresmldvn) [inline src/Controller/QuestionTemplateController.php]: strict comparison

File: src/Entity/QuestionTemplate.php [hunk 4]
```diff
@@ -45,6 +46,13 @@ public function setUserID(int $userID) {
         $this->getRepository()->setForeignID($userID, User::class, $this);
     }
 
+    /**
+     * @return EntityInterface
+     */
+    public function getUser() : EntityInterface {
```
Comment (calinpristavu) [inline src/Entity/QuestionTemplate.php]: `getUser()` returns a `User` not an entity interface.

File: src/Entity/QuizTemplate.php [hunk 5]
```diff
@@ -74,4 +75,11 @@ public function getQuestionNumber () {
     public function getQuestions(int $id) {
         return $this->getRepository()->getQuestions($id);
     }
+
+    /**
+     * @return EntityInterface
+     */
+    public function getUser() : EntityInterface {
```
Comment (calinpristavu) [inline src/Entity/QuizTemplate.php]: return type should be concrete, not abstract here.

File: src/Entity/QuestionTemplate.php (hunk 4 above)
Comment (CalinBolea) [inline src/Entity/QuestionTemplate.php]: psr 12

**Pull request merged**